From 05acb953fb145c4f486fc58d2b8d5113d43f8282 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Fri, 29 Mar 2024 16:01:18 +0800
Subject: [PATCH] wifi: mt76: mt7915: fix HE bf capabilities IE for ap and sta
 vif

Without this commit, beamformer and mu_beamformer capabilities are
diabled for ap and sta vif.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt7915/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mt7915/init.c
+++ b/mt7915/init.c
@@ -945,7 +945,7 @@ mt7915_set_stream_he_txbf_caps(struct mt
 	/* the maximum cap is 4 x 3, (Nr, Nc) = (3, 2) */
 	elem->phy_cap_info[7] |= min_t(int, sts - 1, 2) << 3;
 
-	if (vif != NL80211_IFTYPE_AP || vif != NL80211_IFTYPE_STATION)
+	if (!(vif == NL80211_IFTYPE_AP || vif == NL80211_IFTYPE_STATION))
 		return;
 
 	elem->phy_cap_info[3] |= IEEE80211_HE_PHY_CAP3_SU_BEAMFORMER;
